internal/kafka/internal/presenters/admin_kafka_test.go

```diff
@@ -0,0 +1,95 @@
+package presenters
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/api/admin/private"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/api/dbapi"
+	mock "github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/test/mocks/kafkas"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/services/account"
+)
+
+func TestGetRoutesFromKafkaRequest(t *testing.T) {
+	type args struct {
+		dbKafkaRequest *dbapi.KafkaRequest
+	}
+
+	var emptyKafkaRoutes []private.KafkaAllOfRoutes
+
+	tests := []struct {
+		name string
+		args args
+		want []private.KafkaAllOfRoutes
+	}{
+		{
+			name: "should return empty routes without any routes in the KafkaRequest",
+			args: args{
+				dbKafkaRequest: &dbapi.KafkaRequest{},
+			},
+			want: emptyKafkaRoutes,
+		},
+		{
+			name: "should return empty routes if the KafkaRequest routes are malformed",
+			args: args{
+				dbKafkaRequest: mock.BuildKafkaRequest(func(kafkaRequest *dbapi.KafkaRequest) {
+					kafkaRequest.Routes = mock.GetMalformedRoutes()
+				}),
+			},
+			want: emptyKafkaRoutes,
+		},
+		{
+			name: "should return routes if not empty and in correct format in the KafkaRequest",
+			args: args{
+				dbKafkaRequest: mock.BuildKafkaRequest(func(kafkaRequest *dbapi.KafkaRequest) {
+					kafkaRequest.Routes = mock.GetRoutes()
+				}),
+			},
+			want: mock.GetSampleKafkaAllRoutes(),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			routes := GetRoutesFromKafkaRequest(tt.args.dbKafkaRequest)
+			if !reflect.DeepEqual(routes, tt.want) {
+				t.Errorf("unexpected result for GetRoutesFromKafkaRequest - want: %+v got: %+v", tt.want, routes)
+			}
+		})
+	}
+}
+
+func TestPresentKafkaRequestAdminEndpoint(t *testing.T) {
+	type args struct {
+		dbKafkaRequest *dbapi.KafkaRequest
+		accountService account.AccountService
+	}
+
+	tests := []struct {
+		name    string
+		args    args
+		want    *private.Kafka
+		wantErr bool
+	}{
+		{
+			name: "should return admin kafka request",
+			args: args{
+				dbKafkaRequest: mock.BuildKafkaRequest(nil),
+				accountService: account.NewMockAccountService(),
+			},
+			want: mock.BuildAdminKafkaRequest(nil),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			converted, err := PresentKafkaRequestAdminEndpoint(tt.args.dbKafkaRequest, tt.args.accountService)
+			if !tt.wantErr && err != nil {
+				t.Errorf("unexpected error for PresentKafkaRequestAdminEndpoint: %v", err)
+			}
+			if !reflect.DeepEqual(converted, tt.want) {
```

Comment:
Can we use `gomega` and change all these `DeepEqual` checks to something like
```
Expect(converted).To(Equal(tt.want)
```
? (`Equal` internally uses `reflectDeepEqual`)

Reply:
Very good point. I think we should have this suggested as a best practice / team testing style and document it.

Reply:
@machi1990 @ziccardi - I have addressed your comments. do you mind taking another look?